### PR TITLE
Allow contexts with multiple identical devices

### PR DESCRIPTION
**Pull request by nchristensen**
I don't think there is any confusion about which transformations to choose if all of the devices in the context are the same.

### CODE REVIEW

File: src/feinsum/database.py [hunk 1]
```diff
@@ -331,9 +331,12 @@ def query(einsum: FusedEinsum,
...
     if len(cl_ctx.devices) > 1:
-        raise NotImplementedError("CL contexts with multiple devices not supported")
-
-    cl_device, = cl_ctx.devices
+        # Check if all of the devices in the context are the same
+        device_strings = [str(device).split(" at ")[0] for device in cl_ctx.devices]
+        if not all([el == device_strings[0] for el in device_strings]):
+            raise NotImplementedError("CL contexts with multiple device types not supported")
+            
+    cl_device = cl_ctx.devices[0]
```
Comment (kaushikcfd) [inline src/feinsum/database.py:339]: Could be made terser?
```python
    if len({dev.name for dev in cl_ctx.devices}) > 1:
        raise NotImplementedError("CL contexts with multiple devices not supported")

    cl_device = cl_ctx.devices[0]
```

**Review by kaushikcfd — APPROVED** [review]
LGTM, thanks!

File: src/feinsum/database.py (hunk 1 above)
Comment (nchristensen) [reply src/feinsum/database.py:339]: Good idea.

**Comment by kaushikcfd** [comment]
Whoops looks like there were some lint failures. I'll fix those. (The CI was failing on main anyways after some recent loopy updates)

**Pull request merged**